update forge and use op verify (#734)
ethereum-optimism/superchain-ops: src/improvements/NESTED.md

   chain-governor # 0 or 1 or ...
```

> This will generate a file with the name `src/improvements/OPVerifyInputs/<NETWORK_DIR>/<RUNBOOK_DIR>/<TASK_NAME>-input.json`. Keep note of this.

Reviewer: nit, but can we use kebab case like we seem to be using in this repo for directory names > 1 word long.
```suggestion
This will generate a file with the name `src/improvements/op-verify-inputs/<NETWORK_DIR>/<RUNBOOK_DIR>/<TASK_NAME>-input.json`. Keep note of this.
```

Author: fixed

1. Validate integrity of the simulation.
2. Validate correctness of the state diff.

> 3. Validate the domain and message hash with OP Verify.

Reviewer 2: ```suggestion
3. Validate the domain and message hash with `op-txverify`.
```

Reviewer 2: Lots of other places to do this rename, will stop leaving comments on each spot

Author: is this good now?

> 3. Validate and extract domain hash and message hash to approve

Reviewer 3: ```suggestion
3. Validate and extract domain hash and message hash to approve.
```

> 3. Validate, extract domain hash and message hash to approve and compare the output with the outputs of the OP Verify tool.

Reviewer: Is this correct? 

Shouldn't it be something like: 

"Validate the domain and message hash with OP Verify."

Author: fixed

Note down this value. You will need to compare it with the ones
displayed on the Ledger screen at signing.

#### 3.4 Using OP Verify to verify the domain and message hashes

> For this step, we need to clone the OP Verify repository from https://github.com/ethereum-optimism/op-verify if we have not already done so.

Reviewer 4: I think it would be important for signers to clone the op-verify repository well in advance of them needing it. That is because if our github credentials are compromised, `op-verify` could get overwritten with a malicious version during a signing ceremony.

It would probably best for them to clone it during induction, and then again whenever there are significant updates (hopefully very few of these).

Author: is this good to resolve?

```bash
git clone https://github.com/ethereum-optimism/op-verify
```

> Next create a file in the `op-verify` directory called `<output file name>.json` and copy-paste the contents of the `<output file name>.json` file created by the simulation above to it. Then run

Reviewer 4: This is unclear.

The file created from the simulation was `src/improvements/op-verify-inputs/<NETWORK_DIR>/<RUNBOOK_DIR>/<TASK_NAME>-input.json`, am I creating another file with a name of my choosing in `op-verify` and copying the contents of `<TASK_NAME>-input.json` to it? 

Why not just copy `<TASK_NAME>-input.json` into the `op-verify` directory, or running `op-verify` with the full path to `<TASK_NAME>-input.json`?

Also, using `input` and `output` in filenames is confusing here, because they are output of some process and input of another, better to choose a descriptor such as `transaction.json`

Reviewer 3: Honestly I would recommend that we simply kill the creation of this file entirely and focus on the QR verification mode. Much simpler for end-users who don't have to think about multiple potential ways to verify things.

Author: okay, great points! which approach am I to take?

Reviewer 4: I would say to take the QR approach, it is simpler:
 - The user goes to https://op-verify.netlify.app on their phone and enter the safeTxHash from the governance proposal, I guess from gov.optimism.io or VALIDATIONS.md. Their phone displays the QR code.
 - The user runs `op-verify qr` on their signing laptop and have it read the QR code displayed on their phone. Done.

Author: is this good now?

```bash

> go run ./cmd/op-verify --tx <file name>.json

Author: fixed

```

> This will output details about the transaction, including the domain and message hashes. **Compare these to the domain and message hashes you noted down earlier to check if the OP Verify tool is working correctly.**

Author: fixed
